R/derive_extreme_event.R

```diff
@@ -252,6 +380,37 @@
 #'     new_vars = exprs(TRTSDT)
 #'   )
 #'
+#' @info Since the CBOR derivation contains multiple complex parts, it's
+#'   convenient to make use of the `description` argument within each event object
+#'   to describe what condition is being checked.
+#'
+#'   - For the Confirmed Response (CR), for each `"CR"` record in the original ADRS
+#'     dataset that will be identified by the first part of the `condition` argument
+#'     (`AVALC == "CR"`) we need to use the `first_cond_upper` argument to limit the
+#'     group of observations to consider alongside it. Namely, we need to look up to
+#'     and including the second CR (`AVALC.join == "CR"`) over 28 days from the first
+#'     one (`ADT.join >= ADT + 28`). The observations satisfying `first_cond_upper`
+#'     then form part of our "join group", meaning that the remaining portions of
+#'     `condition` which reference joined variables are limited to this group.
+#'     In particular, within `condition` we use `all()` to check that all observations
+#'     are either `"CR"` or `"NE"`, and `count_vals()` to ensure at most one is
+#'     `"NE"`. Note that here we have selected `join_type = "after"` because
+#'     <insert explanation here, explaining why join_type = "all" changes patient 1 to a "PR">.
```

Comment:
Definitely best @bundfussr provides this text

Reply:
i'll approve as from my side i'm all good with the PR, but of course this bit needs to be updated before merging

Reply:
I think i've got my head around this now - i've added an explanation if you'd like to check @bundfussr @rossfarrugia